File: dotnet/net6/Watches-v6/NoAlarm.cs
namespace Watches;

public class NoAlarm : IAlarmStrategy
{
    public void AddAlarm(Alarm alarm)
    {
        Console.WriteLine("Alarms not supported");
    }

    public IEnumerable<Alarm> GetAlarms(DateTime? @from = null, DateTime? to = null)
    {
        return new List<Alarm>();
    }

    public void RemoveAlarm(Alarm alarm)
    {
        Console.WriteLine("Alarms not supported");
    }

    public Alarm? FindAlarmByName(string alarmName)
    {
        return null;
    }
}

File: java/src/com/infoback/NoAlarm.java
package com.infoback;

import java.time.LocalDateTime;
import java.util.ArrayList;

public class NoAlarm implements IAlarmStrategy {
  @Override
  public void addAlarm(Alarm alarm) {
    System.out.println("Alarms not supported");
  }

  @Override
  public Iterable<Alarm> getAlarms(LocalDateTime fromTime, LocalDateTime toTime) {
    return new ArrayList<Alarm>();
  }

  @Override
  public void removeAlarm(Alarm alarm) {
    System.out.println("Alarms not supported");
  }

  @Override
  public Alarm findAlarmByName(String alarmName) {
    return null;
  }
}
